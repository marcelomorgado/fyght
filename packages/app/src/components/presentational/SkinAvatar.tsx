import React from "react";

// Note: https://en.parceljs.org/module_resolution.html#glob-file-paths
import images from "../../assets/img/*.png";

type Props = {
  skin: string; // TODO: To enum
  size: string;
};

export const SkinAvatar = ({ skin, size }: Props) => {
  let px = null;

  if (size === "small") px = 100;
  if (size === "medium") px = 200;

  return <img alt={skin} src={images[skin]} width={px} />;
};
