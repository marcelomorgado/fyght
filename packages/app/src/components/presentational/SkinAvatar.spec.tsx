import * as React from "react";
import { shallow } from "enzyme";

import { SkinAvatar } from "./SkinAvatar";

describe("SkinAvatar", () => {
  test("should render the component", () => {
    expect(shallow(<SkinAvatar size="small" skin=Skin.NINJA />)).toMatchSnapshot();
  });
});
