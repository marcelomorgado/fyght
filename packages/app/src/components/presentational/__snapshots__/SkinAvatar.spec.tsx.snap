// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkinAvatar should render the component 1`] = `
<img
  alt="ninja"
  width={100}
/>
`;
